monai/transforms/io/array.py

```diff
@@ -227,125 +227,152 @@ def __call__(self, filename: Union[Sequence[PathLike], PathLike], reader: Option
         scale: Optional[int] = None,
         dtype: DtypeLike = np.float64,
-        output_dtype: DtypeLike = np.float32,
         squeeze_end_dims: bool = True,
         data_root_dir: PathLike = "",
         separate_folder: bool = True,
         print_log: bool = True,
+        output_format: str = "",
+        writer: Optional[image_writer.ImageWriter] = None,
     ) -> None:
-        self.saver: Union[NiftiSaver, PNGSaver]
-        if output_ext in {".nii.gz", ".nii"}:
-            self.saver = NiftiSaver(
-                output_dir=output_dir,
-                output_postfix=output_postfix,
-                output_ext=output_ext,
-                resample=resample,
-                mode=GridSampleMode(mode),
-                padding_mode=padding_mode,
-                dtype=dtype,
-                output_dtype=output_dtype,
-                squeeze_end_dims=squeeze_end_dims,
-                data_root_dir=data_root_dir,
-                separate_folder=separate_folder,
-                print_log=print_log,
-            )
-        elif output_ext == ".png":
-            self.saver = PNGSaver(
-                output_dir=output_dir,
-                output_postfix=output_postfix,
-                output_ext=output_ext,
-                resample=resample,
-                mode=InterpolateMode(mode),
-                scale=scale,
-                data_root_dir=data_root_dir,
-                separate_folder=separate_folder,
-                print_log=print_log,
-            )
-        else:
-            raise ValueError(f"unsupported output extension: {output_ext}.")
+        self.folder_layout = FolderLayout(
+            output_dir=output_dir,
+            postfix=output_postfix,
+            extension=output_ext,
+            parent=separate_folder,
+            makedirs=True,
+            data_root_dir=data_root_dir,
+        )
+
+        self.output_ext = output_ext.lower() or output_format.lower()
+        self.writers = image_writer.resolve_writer(self.output_ext) if writer is None else (writer,)
+        self.writer_obj = None
+
+        _output_dtype = output_dtype
+        if self.output_ext == ".png" and _output_dtype not in (np.uint8, np.uint16):
+            _output_dtype = np.uint8
+        if self.output_ext == ".dcm" and _output_dtype not in (np.uint8, np.uint16):
+            _output_dtype = np.uint8
+        self.init_kwargs = {"output_dtype": _output_dtype, "scale": scale}
+        self.data_kwargs = {"squeeze_end_dims": squeeze_end_dims, "channel_dim": 0}
+        self.meta_kwargs = {"resample": resample, "mode": mode, "padding_mode": padding_mode, "dtype": dtype}
+        self.write_kwargs = {"verbose": print_log}
+        self._data_index = 0
+
+    def set_options(self, init_kwargs=None, data_kwargs=None, meta_kwargs=None, write_kwargs=None):
+        """
+        Set the options for the underlying writer by updating `kwargs` dictionaries.
```

Comment:
Maybe it can be more clear: updating the default `kwargs` which were generated in the constructor?

Thanks.